src/kubectlutils.ts

```diff
@@ -17,157 +17,123 @@
+export async function getConfigMaps(): Promise<string[]> {
+	const namespace: string | undefined = config.getNamespaceconfig();
 	if (namespace) {
-		return getNamedListFromKubernetesThenParseList('configmap', `--namespace=${namespace}`);
+		return await getNamedListFromKubernetesThenParseList('configmap', `--namespace=${namespace}`);
 	} else {
-		return getNamedListFromKubernetesThenParseList('configmap');
+		return await getNamedListFromKubernetesThenParseList('configmap');
 	}
 }
 
-export function getSecrets(): Promise<string[]> {
-	let namespace: string | undefined = config.getNamespaceconfig();
+export async function getSecrets(): Promise<string[]> {
+	const namespace: string | undefined = config.getNamespaceconfig();
 	if (namespace) {
-		return getNamedListFromKubernetesThenParseList('secret', `--namespace=${namespace}`);
+		return await getNamedListFromKubernetesThenParseList('secret', `--namespace=${namespace}`);
 	} else {
-		return getNamedListFromKubernetesThenParseList('secret');
+		return await getNamedListFromKubernetesThenParseList('secret');
 	}
 }
 
-export function getIntegrations(): Promise<string> {
-	let namespace: string | undefined = config.getNamespaceconfig();
+export async function getIntegrations(): Promise<string> {
+	const namespace: string | undefined = config.getNamespaceconfig();
 	if (namespace) {
-		return getNamedListFromKubernetes('integration', `--namespace=${namespace}`);
+		return await getNamedListFromKubernetes('integration', `--namespace=${namespace}`);
 	} else {
-		return getNamedListFromKubernetes('integration');
+		return await getNamedListFromKubernetes('integration');
 	}
 }
 
-export function getPodsFromKubectlCli() : Promise<string> {
-	return new Promise<string>( async (resolve, reject) => {
-		await k8s.extension.kubectl.v1
-			.then( async (kubectl) => {
-				let cmd = `get pods`;
-				if (kubectl && kubectl.available) {
-					return await kubectl.api.invokeCommand(cmd);
-				} else {
-					reject(new Error('Kubernetes not available'));
-				}
-			})
-			.then( (result) => {
-				if (!result || result.code !== 0) {
-					let error = `Unable to invoke kubectl to retrieve pod information`;
-					if (result && result.stderr) {
-						error = result.stderr;
-					}
-					reject(error);
-				} else if (result) {
-					const splitResults = result.stdout;
-					resolve(splitResults);
-				}
-			})
-			.catch( (err) => reject(err) );
-	});
+export async function getPodsFromKubectlCli() : Promise<string> {
+	const kubectl: k8s.API<k8s.KubectlV1> = await k8s.extension.kubectl.v1;
+	const cmd: string = `get pods`;
+	if (kubectl && kubectl.available) {
+		const result: k8s.KubectlV1.ShellResult | undefined = await kubectl.api.invokeCommand(cmd);
+		if (!result || result.code !== 0) {
+			let error: string = `Unable to invoke kubectl to retrieve pod information`;
+			if (result && result.stderr) {
+				error = result.stderr;
+			}
+			return error;
```

Comment:
shouldn't we throw an error instead of returning it as a string like if everything went fine?